recipe/meta.yaml

```diff
@@ -11,27 +14,102 @@ source:
+      host:
+        - {{ pin_subpackage('libfreetype' ~ so_name_major, exact=True) }}        # [unix]
+      run:
+        - {{ pin_subpackage('libfreetype' ~ so_name_major, exact=True) }}        # [unix]
+    test:
+      commands:
+        - ${PREFIX}/bin/freetype-config --version  # [not win]
+      # Need these things
+        - test -f ${PREFIX}/lib/libfreetype{{ SHLIB_EXT }}                       # [unix]
+        - test -f ${PREFIX}/lib/libfreetype.{{ so_name_major }}{{ SHLIB_EXT }}   # [osx]
+        - test -f ${PREFIX}/lib/libfreetype{{ SHLIB_EXT }}.{{ so_name_major }}   # [linux]
+        - test -f ${PREFIX}/lib/libfreetype{{ SHLIB_EXT }}.{{ so_name }}         # [linux]
+        - if not exist %PREFIX%\\Library\\lib\\freetype.lib exit 1               # [win]
+        - test -f ${PREFIX}/include/freetype2/freetype/freetype.h                # [unix]
+        - if not exist %PREFIX%\\Library\\include\\freetype\\freetype.h exit 1   # [win]
+        - test -f ${PREFIX}/lib/pkgconfig/freetype2.pc                            # [unix]
+        - if not exist %PREFIX%\\Library\\lib\\pkgconfig\\freetype2.pc exit 1     # [win]
+        - if not exist %PREFIX%\\Library\\lib\\cmake\\freetype\\freetype-config.cmake exit 1     # [win]
+        - test -f ${PREFIX}/bin/freetype-config                               # [unix]
+      # Exclude these things
+        - test ! -f ${PREFIX}/lib/libfreetype.a                               # [unix]
+        - if exist %PREFIX%\\Library\\bin\\freetype.dll exit 1                # [win]
+
+  # Only what's needed for run_exports downstream
+  - name: libfreetype{{ so_name_major }}
+    files:
+      include:
+        - "Library/bin/*.dll"  # [win]
+        - "lib/lib*.*.dylib"  # [osx]
+        - "lib/lib*.so.*"  # [linux]
+    build:
+      run_exports:
+        - {{ pin_subpackage('libfreetype' ~ so_name_major) }}
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ stdlib('c') }}
+      host:
+        - libpng
+        - zlib
+      run:
+        - {{ pin_subpackage('_libfreetype_api', max_pin='x.x.x') }}              # [win]
+      # Pre-split versions of freetype don't know about libfreetypeX, so constrain
+      # them from being installed to prevent clobber.
+      run_constrained:
+        - {{ pin_subpackage('freetype', min_pin='x.x.x') }}
+    test:
+      commands:
+      # Need these things
+        - test -f ${PREFIX}/lib/libfreetype.{{ so_name_major }}{{ SHLIB_EXT }}   # [osx]
+        - test -f ${PREFIX}/lib/libfreetype{{ SHLIB_EXT }}.{{ so_name_major }}   # [linux]
+        - test -f ${PREFIX}/lib/libfreetype{{ SHLIB_EXT }}.{{ so_name }}         # [linux]
+        - if not exist %PREFIX%\\Library\\bin\\freetype.dll exit 1               # [win]
+
+  # Clobber preventer for Windows; prevents multiple freetype.dll files from being
+  # installed simultaneously. This lets us avoid exporting a strict pin on the
+  # libfreetype API which is not strictly correlated with the ABI.
```

Comment:
Do you really want to allow co-installability of different libfreetype ABI on Unix? It's better to not allow that.

Reply:
Probably I agree. I plan to make more changes to this recipe based on discussion in https://github.com/conda-forge/conda-forge.github.io/issues/2401